engine/src/main/java/com/datatorrent/stram/client/AppPackage.java

```diff
@@ -426,7 +452,10 @@ private void processPropertiesXml(File file, AppInfo app)
       config.loadFile(file);
       for (Map.Entry<String, String> entry : config) {
         String key = entry.getKey();
+        PropertyInfo propertyInfo = new PropertyInfo();
```

Comment:
It is not necessary to construct propertyInfo when `value == null`, move to line 466.